clientconn.go

```diff
@@ -362,20 +362,23 @@ func chainUnaryClientInterceptors(cc *ClientConn) {
 	} else if len(interceptors) == 1 {
 		chainedInt = interceptors[0]
 	} else {
-		chainedInt = func(ctx context.Context, method string, req, reply interface{}, cc *ClientConn, invoker UnaryInvoker, opts ...CallOption) error {
-			return interceptors[0](ctx, method, req, reply, cc, getChainUnaryInvoker(interceptors, 0, invoker), opts...)
-		}
+		chainedInt = chainManyUnaryClientInterceptors(interceptors)
 	}
 	cc.dopts.unaryInt = chainedInt
 }
 
-// getChainUnaryInvoker recursively generate the chained unary invoker.
-func getChainUnaryInvoker(interceptors []UnaryClientInterceptor, curr int, finalInvoker UnaryInvoker) UnaryInvoker {
-	if curr == len(interceptors)-1 {
-		return finalInvoker
-	}
-	return func(ctx context.Context, method string, req, reply interface{}, cc *ClientConn, opts ...CallOption) error {
-		return interceptors[curr+1](ctx, method, req, reply, cc, getChainUnaryInvoker(interceptors, curr+1, finalInvoker), opts...)
+func chainManyUnaryClientInterceptors(interceptors []UnaryClientInterceptor) UnaryClientInterceptor {
```

Comment:
Nit: Drop "Many" from these function names. GoLang style leans toward Brevity.

Reply:
I wasn't able to come up with a right name since all function names like `chain(Client|Server)(Unary|Stream)Interceptors` are already in use and `chain(Unary|Stream)Interceptors` belonged to the server side, so to avoid confusion I simply inlined multi-interceptor chaining in all related functions and replaced `else if` branches with `switch case`s.